python-checks/src/main/java/org/sonar/python/checks/FlaskHardCodedSecret.java

```diff
@@ -117,40 +115,40 @@ private static boolean isCallToDictConstructor(CallExpression callExpression) {
       .isPresent();
   }
 
-  private static boolean hasIllegalKeyValuePair(DictionaryLiteral dictionaryLiteral) {
+  private boolean hasIllegalKeyValuePair(DictionaryLiteral dictionaryLiteral) {
     return dictionaryLiteral.elements().stream()
       .filter(KeyValuePair.class::isInstance)
       .map(KeyValuePair.class::cast)
       .map(KeyValuePair::key)
       .filter(StringLiteral.class::isInstance)
       .map(StringLiteral.class::cast)
       .map(StringLiteral::trimmedQuotesValue)
-      .anyMatch(SECRET_KEY_KEYWORD::equals);
+      .anyMatch(key -> this.getSecretKeyKeyword().equals(key));
   }
 
-  private static boolean hasIllegalKeywordArgument(CallExpression callExpression) {
-    return Optional.ofNullable(TreeUtils.argumentByKeyword(SECRET_KEY_KEYWORD, callExpression.arguments()))
+  private boolean hasIllegalKeywordArgument(CallExpression callExpression) {
+    return Optional.ofNullable(TreeUtils.argumentByKeyword(this.getSecretKeyKeyword(), callExpression.arguments()))
       .map(RegularArgument::expression)
-      .filter(FlaskHardCodedSecretCheck::isStringLiteral)
+      .filter(FlaskHardCodedSecret::isStringLiteral)
       .isPresent();
   }
 
-  private static void verifyAssignmentStatement(SubscriptionContext ctx) {
+  private void verifyAssignmentStatement(SubscriptionContext ctx) {
     AssignmentStatement assignmentStatementTree = (AssignmentStatement) ctx.syntaxNode();
     List<Expression> expressionList = assignmentStatementTree.lhsExpressions().stream()
       .map(ExpressionList::expressions)
       .filter(list -> list.size() == 1)
       .flatMap(List::stream)
-      .filter(FlaskHardCodedSecretCheck::isSensitiveProperty)
+      .filter(this::isSensitiveProperty)
       .filter(expression -> isStringLiteral(assignmentStatementTree.assignedValue()))
```

Comment:
move this check as an `if` condition in the method beginning:
```java
if (!isStringLiteral(assignmentStatementTree.assignedValue())) {
      return;
    }
```

Reply:
Done in 73be1ac3b25eb4b2c9dc41cf27f4f93aa2251763.